### ISSUE TITLE
webservice

### ISSUE THREAD

**Issue by ahmadRagheb** [ageitgey/face_recognition#239]
hi , i spend this week trying to send image form java script throw WebsocketServer  i can send 
            var data = canvas.get()[0].toDataURL('image/png', 1.0);
            ws.send(data);

but don't know how could i pass them to face recongnition api ,, i tried so much things!! ???

**Comment by eloelo22** [comment]
i have done something similar to this a web service that enrolls and compares faces from the enrolled face encodings here is a screenshot of the result from the web service together with possible matches is this some what similar to what you want to do?
![face1](https://user-images.githubusercontent.com/8648535/33190581-16bf912c-d0e9-11e7-9251-4973ddbaeb42.png)

![face2](https://user-images.githubusercontent.com/8648535/33190587-27d52882-d0e9-11e7-8768-35ae6d52abe1.png)

**Comment by ahmadRagheb** [comment]
what i actually need is when you use html5 <video> and <canvas>  to send video frames to Face Recognition API by passing video to canvas then send it as URI to server  .. how could i send these images to the server in a format that API can understand like the same frame it takes from the server cam when you run it locally .  or how could i change the uri to image format in python ,, that API accept ??

**Comment by eloelo22** [comment]
I also had trouble with the web service have you tried using form data?

**Comment by eloelo22** [comment]
Have you tried converting the uri to image in client side? Then sending a form data with the converted uri to image to the web service

**Comment by ahmadRagheb** [comment]
that's what i'm asking for i used many codes but they are not the right color or structure the package 
api takes

**Comment by ahmadRagheb** [comment]
we need like how to convert them at front end and do we need to do code at back end before passing it to api

**Comment by eloelo22** [comment]
Well i cant help you with the conversion I haven’t tried doing so i think its a bit different In dealing with an uploaded photo vs a video frame on my wclient side i just declare a form data then send it to the web service i am sorry if I couldn’t help you with the uri conversion good luck

**Comment by eloelo22** [comment]
Let me share to you what i did on my web service if you pass a formdata with the image there is no need to add additional code the sample web service will suffice unless you want to do face recognition which I suspect is where you are heading

**Comment by ahmadRagheb** [comment]
@eloelo22  thank you very much , know it takes lot of my time but i will solve it soon , and the code will save someone time .

**Comment by eloelo22** [comment]
Sure thing if ever you need a sample code from my web service let me know 😃😃

**Comment by ahmadRagheb** [comment]
i use at frontend 
    timer = setInterval(
        function() {

            ctx.drawImage(video, 0, 0, 640, 480 );
            var data = canvas.get()[0].toDataURL('image/png', 1.0);
            ws.send(data);
        }, 250);

it send this type of data to server : 
data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAoAAAAHgCAYAAAA10dzkAAAgAElEQVR4Xuy9i3ZkN64kqpTkLlfP3P//13uqpFlAIIAASO5M+dHHdqd6uaXK3A8SJIFg4MHb93//+/Nl/NgH/t9nffXz4+Pl8/by8vHx4+Xl5QNXvL2+vLy+v7zc3l5++by9vH7intvL7eX15fXlNp58e33NN33a8+T59sXtdptNWf7Ne+y3P95ecnt5eX2tez8/Xl7sP/vEvn693V5ur7eX203ebxds+m0fWT/96dEB3vcZ93x8oGPez1v109.....

at server side : 

    strOne = 'b'+request
    strOne = strOne.partition(",")[2]
    pad = len(strOne)%4
    strOne += b"="*pad
    image=base64.decodestring(strOne.strip())


don't know what to do next .  i need to make this image numppy array that accepted by face 
face_recognition.face_encodings
face_recognition.face_locations

**Comment by stoneWeb** [comment]
@ahmadRagheb So, did you solve it, I also want to know how to use ws transport.

**Comment by ahmadRagheb** [comment]
@stoneWeb  yes i use it inside a framework called frappe python /java script framework 
https://github.com/ahmadRagheb/facefrappe

**Comment by ahmadRagheb** [comment]
https://discuss.frappe.io/t/frappe-login-based-on-face-recognition-and-detection/496
